platform-sdk/swirlds-common/src/main/java/com/swirlds/common/merkle/proof/algorithms/StateProofConstants.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.merkle.proof.algorithms;
+
+import static com.swirlds.common.units.DataUnit.UNIT_BYTES;
+import static com.swirlds.common.units.DataUnit.UNIT_MEGABYTES;
+
+/**
+ * Constants used by the state proof implementation.
+ */
+public final class StateProofConstants {
+
+    /**
+     * The maximum number of signatures supported by deserialization.
+     */
+    public static final int MAX_SIGNATURE_COUNT = 1024;
```

Comment:
Is this count tied to other count values?  Is there a way to note they are tied together?

Reply:
It's not anything special, other than a number that I thought was sufficiently large for the time being. If we allow an unlimited number of signatures, an attacker could send us a state proof with a trillion signatures and cause us to run out of memory.